src/Microsoft.OData.Core/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ServiceCollectionExtensions.cs" company="Microsoft">
+// Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Json;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.OData.Core
+{
+    /// <summary>
+    /// Extension methods for <see cref="IServiceCollection"/>.
+    /// </summary>
+    public static class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds the default OData services to the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> instance itself.</returns>
+        public static IServiceCollection AddDefaultODataServices(this IServiceCollection services)
+        {
+            return AddDefaultODataServices(services, ODataVersion.V4);
+        }
+
+        /// <summary>
+        /// Adds the default OData services to the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <param name="odataVersion">ODataVersion for the default services.</param>
+        /// <returns>The <see cref="IServiceCollection"/> instance itself.</returns>
+        public static IServiceCollection AddDefaultODataServices(this IServiceCollection services, ODataVersion odataVersion)
+        {
+            if (services == null)

```

Comment:
This should be `if (services is null)`. This new syntax is different than an equality check and improves performance.

Reply:
updated